sdk/communication/Azure.Communication.PhoneNumbers/tests/SipRouting/Infrastructure/TestData.cs

```diff
@@ -1,39 +1,51 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
 using System.Collections.Generic;
+using Azure.Core.TestFramework;
 
 namespace Azure.Communication.PhoneNumbers.SipRouting.Tests
 {
-    internal static class TestData
+    public class TestData
     {
-        public static readonly List<string> Fqdns = new List<string>(){ "sbs1.sipconfigtest.com", "sbs2.sipconfigtest.com" };
-        public static readonly int[] TrunkPorts = { 1122, 1123 };
+        private readonly string Domain;
+        public readonly List<string> Fqdns;
+        public readonly int[] TrunkPorts = { 1122, 1123 };
 
-        public static readonly List<SipTrunk> TrunkList = new List<SipTrunk>
-        {
-            new SipTrunk(Fqdns[0], TrunkPorts[0]),
-            new SipTrunk(Fqdns[1], TrunkPorts[1])
-        };
-
-        public static readonly SipTrunk NewTrunk = new SipTrunk("newsbs.sipconfigtest.com", 3333);
+        public readonly List<SipTrunk> TrunkList;
+        public readonly SipTrunk NewTrunk;
 
-        public static readonly SipTrunkRoute RuleNavigateToTrunk1 = new SipTrunkRoute(
-            name: "First rule",
-            description: "Handle numbers starting with '+123'",
-            numberPattern: @"\+123[0-9]+",
-            trunks: new List<string> { Fqdns[0] });
+        public readonly SipTrunkRoute RuleNavigateToTrunk1;
+        public readonly SipTrunkRoute RuleNavigateToAllTrunks;
+        public readonly SipTrunkRoute RuleNavigateToNewTrunk;
 
-        public static readonly SipTrunkRoute RuleNavigateToAllTrunks = new SipTrunkRoute(
-            name: "Last rule",
-            description: "Handle all other numbers'",
-            numberPattern: @"\+[1-9][0-9]{3,23}",
-            trunks: Fqdns);
+        public TestData(RecordedTestMode? mode)
+        {
+            Domain = mode != RecordedTestMode.Playback ? ("." + Guid.NewGuid()) : "";
```

Comment:
Do you want this to be truly random, or recording random? (e.g `Recording.Random.NewGuid()`)

Reply:
Used `Recording.Random.NewGuid()` instead of `Guid.NewGuid()`
Domain sanitizer was removed, since it is no longer needed.

`Guid.NewGuid()` was called directly in the `TestData` constructor.
Now the Guid parameter is passed into the constructor from the test.
(In the test the `Recording` variable is accessible via parent)

`TestData = new TestData(Recording.Random.NewGuid())` is called from a `[SetUp]` method now because the `Recording` variable is not setup until then.